### PR TITLE
Comments | Add optional relevant_date field

### PR DESCRIPTION
**Pull request by dannysellers**
Connects #2983 

## AC
- [ ] After clicking/adding a comment, there should be an optional field for adding a relevant_date
- [ ] Comment with date should be of format [**mm/dd/yyyy** - Text...]
- [ ] If no relevant_date saved, preserve comment
- [ ] Page number should be above the comment box
- [ ] Comments should be sorted by page number

### CODE REVIEW

File: client/app/components/DateSelector.jsx [hunk 1]
```diff
@@ -13,6 +13,11 @@ export default class DateSelector extends React.Component {
     let value = initialValue || '';
     let propsValue = this.props.value || '';
 
+    if (this.props.type === 'date' && this.props.onChange) {
```
Comment (mdbenjam) [inline client/app/components/DateSelector.jsx]: I like what you're doing here. It's a little ugly, but it's a great way to transition off of our own validation!

File: client/app/reader/AnnotationLayer/AnnotationLayerReducer.js [hunk 2]
```diff
@@ -171,6 +171,16 @@ export const annotationLayerReducer = (state = initialState, action = {}) => {
         }
       }
     });
+  case Constants.UPDATE_ANNOTATION_RELEVANT_DATE:
```
Comment (mdbenjam) [inline client/app/reader/AnnotationLayer/AnnotationLayerReducer.js]: I wish we could just add one action. But I don't see a clear way to do it.

Comment (dannysellers) [reply client/app/reader/AnnotationLayer/AnnotationLayerReducer.js]: Yeah :(

File: client/app/components/DateSelector.jsx (hunk 1 above)
Comment (dannysellers) [reply client/app/components/DateSelector.jsx]: :+1: It should be pretty easy to transition to `type="date"` in other cases too—we basically wouldn't need any of this validation/input modification any more

**Comment by gnakm** [comment]
@dannysellers Yes, Relevant date should be bold and start about a line below the add comment button. Apologies, didn't add comment button in mock. 

image

**Comment by dannysellers** [comment]
👍

**Pull request merged**